model/config.go

```diff
@@ -223,6 +223,7 @@ type TeamSettings struct {
 	RestrictPublicChannelManagement  *string
 	RestrictPrivateChannelManagement *string
 	UserStatusAwayTimeout            *int64
+	MaxChannelsPerTeam               int
```

Comment:
We're trying to migrate to pointers for all config settings, there's a [ticket](https://mattermost.atlassian.net/browse/PLT-3385) for it